extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/indexing/CountingOutputChannelFactory.java

```diff
@@ -50,7 +51,22 @@ public OutputChannel openChannel(int partitionNumber) throws IOException
             new CountingWritableFrameChannel(
                 baseChannel.getWritableChannel(),
                 channelCounters,
-                baseChannel.getPartitionNumber()
+                partitionNumber
+            )
+    );
+  }
+
+  @Override
+  public PartitionedOutputChannel openPartitionedChannel(String name, boolean deleteAfterRead) throws IOException
+  {
+    final PartitionedOutputChannel baseChannel = baseFactory.openPartitionedChannel(name, deleteAfterRead);
+
+    return baseChannel.mapWritableChannel(
+        baseWritableChannel ->
+            new CountingWritableFrameChannel(
+                baseChannel.getWritableChannel(),
+                channelCounters,
+                null
```

Comment:
why is empty partition Number passed here ?

Reply:
It is passed because the channel can contain multiple partitions, so the partition number is looked upon every frame being written